domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java

```diff
@@ -971,7 +973,12 @@ private static void streamTrailer(HttpServerResponse response, ResultInfo result
       PostgresClient postgresClient = postgresClient(vertxContext, okapiHeaders);
       postgresClient.update(table, entity, id, reply -> {
         if (reply.failed()) {
-          asyncResultHandler.handle(response(table, id, reply.cause(), clazz, respond400, respond500));
+          if (PgExceptionUtil.isVersionConflict(reply.cause())) {
+            Method method = respond409 == null ? respond400 : respond409;
+            asyncResultHandler.handle(response(reply.cause().getMessage(), method, method));
```

Comment:
```suggestion
            asyncResultHandler.handle(response(reply.cause().getMessage(), method, respond500));
```
If creating a respond400 or respond409 throws an exception we need to fall-back to respond500.

Reply:
> If creating a respond400 or respond409 throws an exception we need to fall-back to respond500.

Done.